stdlib/Distributed/src/macros.jl

```diff
@@ -260,9 +260,27 @@ function splitrange(N::Int, np::Int)
     return chunks
 end
 
+# Statically split range [firstIndex,lastIndex] into equal sized chunks for np processors
+function splitrange(firstIndex::Int, lastIndex::Int, np::Int)
+    each = div(lastIndex-firstIndex+1,np)
+    extras = rem(lastIndex-firstIndex+1,np)
+    nchunks = each > 0 ? np : extras
+    chunks = Vector{UnitRange{Int}}(undef, nchunks)
+    lo = firstIndex
+    for i in 1:nchunks
+        hi = lo + each - 1
+        if extras > 0
+            hi += 1
+            extras -= 1
+        end
+        chunks[i] = lo:hi
+        lo = hi+1
+    end
+    return chunks
+end
+
 function preduce(reducer, f, R)
-    N = length(R)
-    chunks = splitrange(Int(N), nworkers())
+    chunks = splitrange(firstindex(R),lastindex(R), nworkers())
```

Comment:
```suggestion
    chunks = splitrange(Int(firstindex(R)), Int(lastindex(R)), nworkers())
```

Please also generally use `, ` (comma and space) consistently between arguments.